### ISSUE TITLE
Updating .env and Changing Model Configurations

### ISSUE THREAD

**Issue by DSOTM-RSA** [Azure-Samples/chat-with-your-data-solution-accelerator#1306]
# Describe the bug
Having succesfully deployed the solution via Local Development/Deployment I would like to change my **.env** to reflect am alternate configruation, i.e. GPT3.5 to GPT4o for example. Adjusting the .env as needed, I receive the following error.

![image](https://github.com/user-attachments/assets/ad802970-7d59-4701-a61d-6028ff2620f8)

It seems like I can't update the **open-ai service** to use an alternative (chat) model.

# Expected behavior
The ability to change .env to deploy new models, or use existing models (in the same azure-openai instance) for previously deployed azd environments. _[I've done this many times for the alternative [ChatGPT + Enterprise data with Azure OpenAI and AI Search demo](https://github.com/Azure-Samples/azure-search-openai-demo) successfully]_

How can I change, configure my deplyoments to meet/match the case as presented in the contract existence example, or in any custom configuration of my choice, gpt4-o and embedding-small etc.

Am I missing something fundamentally different about how these two solutions use azd, change existing/create new enviornments or use existing resources??

# Debugging information
Windows 10 >> WSL >> Dev Container
RG = swedencentral

## Logs

If applicable, add logs to help the engineer debug the problem.

---

# Tasks

_To be filled in by the engineer picking up the issue_

- [ ] Task 1
- [ ] Task 2
- [ ] ...

**Comment by Roopan-Microsoft** [comment]
Hey @DSOTM-RSA , Thanks for your feedback. If you want to update the GPT Model. You have 2 options.
- When you do One Click Deployment, you have a provision to update the GPT model and do the deployment
- If you are trying local deployment, you can update by following some steps.

Will provide you the updates shortly.

Thank you for your patience. I really appreciate it.

**Comment by Roopan-Microsoft** [comment]
Hey @DSOTM-RSA - We figured out the issue that few of the env variables are not getting populated in .env file.

We are currently working on that. Will keep you posted. Thanks

**Comment by Pavan-Microsoft** [comment]
Hi @DSOTM-RSA - Thank you for your patience. We have identified the issue regarding the environment variables not being populated in the .env file. Due to a limitation in the Bicep output variable, which can only accommodate 64 variables, we were unable to include the missing variables individually.
 
To address this, we have consolidated the OpenAI model variables and the Azure OpenAI embedding model into a single object. This change allows us to update the .env file to deploy new models or utilize existing models within the same Azure OpenAI instance for previously deployed Azure Development environments.
 
Previously, the variable AZURE_OPENAI_MODEL was missing; however, we have now consolidated the OpenAI model information into the AZURE_OPENAI_MODEL_INFO object, which includes AZURE_OPENAI_MODEL, AZURE_OPENAI_MODEL_NAME, and AZURE_OPENAI_MODEL_VERSION.
<img width="791" alt="github2709" src="https://github.com/user-attachments/assets/b8b2a36e-aa7b-43fa-9695-b89645739343">


We appreciate your understanding. Thank you!

CC: @Roopan-Microsoft
